### PR TITLE
Replace Enum.Parse() by pre-built dictionary

### PR DESCRIPTION
**Pull request by SergeiPavlov**

### CODE REVIEW

File: Orm/Xtensive.Orm/Orm/Model/Stored/Internals/ReferenceUpdater.cs [hunk 1]
```diff
@@ -14,6 +14,15 @@ namespace Xtensive.Orm.Model.Stored
 {
   internal class ReferenceUpdater
   {
+    private static readonly Dictionary<string, Multiplicity> MultiplicityMap = Enum.GetValues<Multiplicity>().ToDictionary(o => o.ToString());
```
Comment (snaumenko-st) [inline Orm/Xtensive.Orm/Orm/Model/Stored/Internals/ReferenceUpdater.cs]:
```suggestion
    private static readonly FrozenDictionary<string, Multiplicity> MultiplicityMap = Enum.GetValues<Multiplicity>().ToFrozenDictionary(o => o.ToString());
```

File: Orm/Xtensive.Orm/Orm/Linq/Translator.Queryable.cs [hunk 2]
```diff
@@ -27,6 +27,8 @@ namespace Xtensive.Orm.Linq
 {
   internal sealed partial class Translator : QueryableVisitor
   {
+    private static readonly Dictionary<string, ResultAccessMethod> ResultAccessMethodMap = Enum.GetValues<ResultAccessMethod>().ToDictionary(o => o.ToString());
```
Comment (snaumenko-st) [inline Orm/Xtensive.Orm/Orm/Linq/Translator.Queryable.cs]:
```suggestion
    private static readonly FrozenDictionary<string, ResultAccessMethod> ResultAccessMethodMap = Enum.GetValues<ResultAccessMethod>().ToFrozenDictionary(o => o.ToString());
```

File: Orm/Xtensive.Orm/Orm/Model/Stored/Internals/ReferenceUpdater.cs [hunk 3]
```diff
@@ -14,6 +14,15 @@ namespace Xtensive.Orm.Model.Stored
...
+
+    private static readonly Dictionary<string, InheritanceSchema> InheritanceSchemaMap = new() {
+      [nameof(InheritanceSchema.Default)] = InheritanceSchema.Default,
+      [nameof(InheritanceSchema.ClassTable)] = InheritanceSchema.ClassTable,
+      [nameof(InheritanceSchema.SingleTable)] = InheritanceSchema.SingleTable,
+      [nameof(InheritanceSchema.ConcreteTable)] = InheritanceSchema.ConcreteTable
+    };
```
Comment (snaumenko-st) [inline Orm/Xtensive.Orm/Orm/Model/Stored/Internals/ReferenceUpdater.cs]:
```suggestion
    private static readonly FrozenDictionary<string, InheritanceSchema> InheritanceSchemaMap = new Dictionary<string, InheritanceSchema> {
      [nameof(InheritanceSchema.Default)] = InheritanceSchema.Default,
      [nameof(InheritanceSchema.ClassTable)] = InheritanceSchema.ClassTable,
      [nameof(InheritanceSchema.SingleTable)] = InheritanceSchema.SingleTable,
      [nameof(InheritanceSchema.ConcreteTable)] = InheritanceSchema.ConcreteTable
    }.ToFrozenDictionary();
```

**Review by snaumenko-st — COMMENTED** [review]
`FrozenDictionary` performs better in this case
| Method              | Mean       | Error    | StdDev   | Ratio | RatioSD | Gen0    | Allocated | Alloc Ratio |
|-------------------- |-----------:|---------:|---------:|------:|--------:|--------:|----------:|------------:|
| ParseEnum           | 2,493.5 μs | 48.65 μs | 99.38 μs |  1.00 |    0.00 | 74.2188 |  960039 B |       1.000 |
| ParseEnumCachedType | 2,382.9 μs | 46.63 μs | 68.35 μs |  0.95 |    0.05 | 74.2188 |  960040 B |       1.000 |
| ParseEnumGeneric    |   685.9 μs | 13.63 μs | 23.51 μs |  0.27 |    0.02 |       - |       1 B |       0.000 |
| Dictionary       |   479.8 μs |  9.57 μs | 19.34 μs |  0.19 |    0.01 |       - |       1 B |       0.000 |
| FrozenDict     |   195.6 μs |  3.73 μs |  7.10 μs |  0.08 |    0.00 |       - |         - |       0.000 |

**Review by botinko — APPROVED** [review]

**Pull request merged**